pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -493,3 +508,34 @@ func (p *Provisioner) populateDomainAndPort(sc *storagev1.StorageClass) error {
 
 	return nil
 }
+
+// Check for quota limit in OBC and set accordingly
+func (p Provisioner) setAdditionalSettings(options *apibkt.BucketOptions) error {
+	maxObjects := MaxObjectQuota(options)
+	maxSize := MaxSizeQuota(options)
+	var err error
+
+	if maxObjects != "" {
+		_, err = cephObject.SetQuotaUserObjectMax(p.objectContext, p.cephUserName, maxObjects)
+		if err != nil {
+			logger.Errorf("Setting MaxObject failed %v", err)
+			return err
+		}
+	}
+	if maxSize != "" {
+		_, err = cephObject.SetQuotaUserMaxSize(p.objectContext, p.cephUserName, maxSize)
+		if err != nil {
+			logger.Errorf("Setting MaxSize failed %v", err)
+			return err
+		}
+	}
+	if maxObjects != "" || maxSize != "" {
+		_, err = cephObject.EnableUserQuota(p.objectContext, p.cephUserName)
+		if err != nil {
+			logger.Errorf("Enabling user quota for obc failed %v", err)
+			return err
+		}
+	}
+
+	return nil
+}
```

Comment:
This is a nit, but you _may_ like this refactor better:
```
func (p Provisioner) setAdditionalSettings(options *apibkt.BucketOptions) error {
	maxObjects := MaxObjectQuota(options)
	maxSize := MaxSizeQuota(options)
        if maxObjects == "" && maxSize == "" {
             return nil
        }

	var err error
	if maxObjects != "" {
		if _, err = cephObject.SetQuotaUserObjectMax(p.objectContext, p.cephUserName, maxObjects); err != nil {
			logger.Errorf("Setting MaxObject failed %v", err)
			return err
		}
	}
	if maxSize != "" {
		if _, err = cephObject.SetQuotaUserMaxSize(p.objectContext, p.cephUserName, maxSize); err != nil {
			logger.Errorf("Setting MaxSize failed %v", err)
			return err
		}
	}
	if _, err = cephObject.EnableUserQuota(p.objectContext, p.cephUserName); err != nil {
		logger.Errorf("Enabling user quota for obc failed %v", err)
	}
	return err
}
 9  pkg/operator/ceph/object/bucket/util.go 
```

Reply:
will address in next patch set